addons/web/static/src/core/debug/profiling/profiling_service.js

```diff
@@ -0,0 +1,82 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { core } = owl;
+
+const profilingService = {
+    dependencies: ["orm"],
+    start(env, { orm }) {
+        const sessionInfo = odoo.session_info;
+        const state = {
+            session: sessionInfo.profile_session || false,
+            collectors: sessionInfo.profile_collectors || ["sql", "traces_async"],
+            params: sessionInfo.profile_params || {},
+            get isEnabled() {
+                return Boolean(state.session);
+            },
+        };
+        const bus = new core.EventBus();
+
+        let recordingIcon = null;
+        function updateDebugIcon() {
+            const debugItem = document.querySelector(".o_main_navbar .o_debug_manager .fa-bug");
+            if (state.isEnabled) {
+                recordingIcon = document.createElement("span");
+                recordingIcon.classList.add("o_recording", "text-danger", "fa", "fa-circle");
+                debugItem.appendChild(recordingIcon);
+            } else if (recordingIcon) {
+                debugItem.removeChild(recordingIcon);
+                recordingIcon = null;
+            }
+        }
+
+        if (env.debug) {
+            env.bus.on("WEB_CLIENT_READY", null, updateDebugIcon);
+        }
+
+        async function setProfiling(params) {
+            const kwargs = Object.assign(
```

Comment:
you can also used args instead of kwargs:
```js
const {isEnabled: profile, collectors, params} = state;
const resp = await orm.call("ir.profile", "set_profiling", [profile, collectors, params];
```